tests/integration-tests/tests/update/test_update.py

```diff
@@ -1230,3 +1230,100 @@ def _test_shared_storage_rollback(
+    #  q2 with 1 subnet in one AZ
+
+    resource_groups_client = boto3.client("resource-groups")
+    odcr_resources = retrieve_cfn_resources(odcr_stack.name, region)
+    resource_group_arn = resource_groups_client.get_group(Group=odcr_stack.cfn_resources["multiAzOdcrGroup"])["Group"][
+        "GroupArn"
+    ]
+
+    init_config_file = pcluster_config_reader(
+        config_file="pcluster_create.config.yaml",
+        multi_az_capacity_reservation_arn=resource_group_arn,
+    )
+
+    cluster = clusters_factory(init_config_file)
+
+    # Retrieves list of cluster instances and checks that
+    # at least one instance was launched in each AZ/Reservation
+    instances = describe_cluster_instances(cluster.name, region, filter_by_compute_resource_name="compute-resource-1")
+    _assert_instance_in_capacity_reservation(instances, odcr_resources["az1Odcr"])
+    _assert_instance_in_capacity_reservation(instances, odcr_resources["az2Odcr"])
+
+    # ## First update
+    # - add a subnet in Queue2
+    # update should succeed without failures or warning messages
+    first_update_config = pcluster_config_reader(
+        config_file="pcluster_update_1.config.yaml",
+        multi_az_capacity_reservation_arn=resource_group_arn,
+    )
+
+    response = cluster.update(str(first_update_config))
+    assert_that(response["clusterStatus"]).is_equal_to("UPDATE_COMPLETE")
+
+    # Second update
+    # - remove a subnet in Queue2
+    # update should fail asking to stop the fleet
+    second_update_config = pcluster_config_reader(
+        config_file="pcluster_update_2.config.yaml",
+        multi_az_capacity_reservation_arn=resource_group_arn,
+    )
+
+    response = cluster.update(str(second_update_config), raise_on_error=False)
+    assert_that(response["message"]).is_equal_to("Update failure")
+    assert_that(response["updateValidationErrors"][0]["message"]).contains("All compute nodes must be stopped")
+
+    # Third update
+    #  - stops the fleet
+    #  - wait until all compute instances are terminated
+    # after fully stopping the fleet, the update should succeed
+
+    cluster.stop()
+    _wait_until_instances_are_stopped(cluster, 360)
+
+    response = cluster.update(str(second_update_config))
+    assert_that(response["computeFleetStatus"]).is_equal_to("STOPPED")
+    assert_that(response["clusterStatus"]).is_equal_to("UPDATE_COMPLETE")
+
+
+def _assert_instance_in_capacity_reservation(instances, expected_reservation):
+    if any(
+        instance["CapacityReservationId"] == expected_reservation
+        for instance in instances
+        if "CapacityReservationId" in instance
+    ):
+        logging.info("Instances found in reservation: %s", expected_reservation)
+    else:
+        logging.error("No instances found in reservation: %s", expected_reservation)
+        pytest.fail("No instances found in the reservation")
+
+
+def _wait_until_instances_are_stopped(cluster, timeout):
```

Comment:
NIT: Have a `step` parameter with `20` as the default value.

```
time.sleep(step)
waited += step
```

Reply:
Alternatively you can leverage [retry](https://pypi.org/project/retry/), the same way it's been used in other tests.